src/main/java/org/openstreetmap/atlas/geography/atlas/delta/AtlasDeltaGenerator.java

```diff
@@ -44,55 +70,134 @@ protected int onRun(final CommandMap command)
     {
         final Path before = (Path) command.get("before");
         final Path after = (Path) command.get("after");
-        final Path outputDir = (Path) command.get("outputDir");
-        run(before, after, outputDir);
+        final Path outputDirectory = (Path) command.get("outputDirectory");
+        threads = (Integer) command.get("threads");
+        run(before, after, outputDirectory);
         return 0;
     }
 
     @Override
     protected SwitchList switches()
     {
-        return new SwitchList().with(BEFORE_SWITCH, AFTER_SWITCH, OUTPUT_DIRECTORY_SWITCH);
+        return new SwitchList().with(BEFORE_SWITCH, AFTER_SWITCH, OUTPUT_DIRECTORY_SWITCH,
+                THREADS_SWITCH);
+    }
+
+    private void run(final Path before, final Path after, final Path outputDirectory)
+    {
+        final Time time = Time.now();
+
+        this.logger.info("Comparing {} and {}", before, after);
+
+        // If the after is a directory, we want to diff the individual shards in parallel.
+        if (Files.isDirectory(after))
+        {
+            // You need to have the before dir be a dir of shards too for this to work.
+            if (!Files.isDirectory(before))
+            {
+                logger.error("Your -before parameter must point to a directory of atlas shards if "
+                        + "you want to compare shard by shard with an -after directory also of shards!");
+                System.exit(COMMAND_LINE_USAGE_ERROR_EXIT);
+            }
+
+            // Execute in a pool of threads so we limit how many atlases get loaded in parallel.
+            final ForkJoinPool customThreadPool = new ForkJoinPool(threads);
+            try
+            {
+                customThreadPool
+                        .submit(() -> this.compareShardByShard(before, after, outputDirectory))
+                        .get();
+            }
+            catch (final InterruptedException interrupt)
+            {
+                logger.error("The shard diff workers were interrupted.", interrupt);
+            }
+            catch (final ExecutionException exec)
+            {
+                logger.error("There was an execution exception on the shard diff workers.", exec);
```

Comment:
`customThreadPool` is never shutdown, recommend doing it in a finally block. Possibly a try-with-resources block.